vault/core.go

```diff
@@ -257,6 +257,12 @@ type Core struct {
 	// token store is used to manage authentication tokens
 	tokenStore *TokenStore
 
+	// identityStore is used to manage client entities
+	identityStore *identityStore
+
+	// entitiesLock is used to ensure thread safety for restoring entities
+	entitiesLock sync.Mutex
```

Comment:
Yes. The entire post unseal operation is happening with the state lock held for write. I will remove the entities lock.